lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1570,8 +1570,8 @@ class ShellCommand : public ExternalCommand {
       return false;
     }
     
-    // Collect the discovered dependencies, if used.
-    if (!depsPaths.empty()) {
+    // Collect the discovered dependencies, if used and command was not cancelled.
+    if (!depsPaths.empty() && !bsci.getDelegate().isCancelled()) {
```

Comment:
I'm not sure I understand why we ever get here -- didn't the command fail if it was cancelled?

I agree that having an enum type to return the status more explicitly is probably appropriate and makes the code more readable.